### PR TITLE
fix jwt cookie name settings not being recognised

### PR DESCRIPTION
**Pull request by nixsiow**
Issue #971 

This should fix the issue by getting the cookie_name from the dj_rest_auth.app_settings instead of Django settings in the original code. I also fix the test case that is associated with this method. I Ran tests locally and everything seems fine from my side. Please let me know if anything needs to be improved. Thanks!

### CODE REVIEW

**Review by tfranzel — CHANGES_REQUESTED** [review]
You got a valid point there but it is only half of the truth. `3.0.0` changed their config mangagement https://github.com/iMerica/dj-rest-auth/pull/477 this was correct prior, and your fix would break people using `pre-3.0.0`.

please adapt the pr to check for the package version and then get the setting from the appropriate location.

**Comment by nixsiow** [comment]
> You got a valid point there but it is only half of the truth. `3.0.0` changed their config mangagement [iMerica/dj-rest-auth#477](https://github.com/iMerica/dj-rest-auth/pull/477) this was correct prior, and your fix would break people using `pre-3.0.0`.
> 
> please adapt the pr to check for the package version and then get the setting from the appropriate location.

Hi there, thanks for your feedback! I amend my commit to check the version in the code. Please let me know if this is adequate or not. This is actually my first time contributing to open source, so please let me know if anything I can improve. Thanks again!

File: drf_spectacular/contrib/rest_auth.py [hunk 1]
```diff
@@ -155,7 +155,24 @@ def get_security_requirement(self, auto_schema):
...
+            # so by assigning a version number (i.e. 1.0.4 in this case)
+            # we can check if it is less than 3.0.0 later on in the code
+            dj_rest_auth_version = "1.0.4"  # 1.0.4 or less in reality
+
+        cookie_name = getattr(api_settings, 'JWT_AUTH_COOKIE', None)
+
+        # fallback for dj_rest_auth version < 3.0.0
+        if int(dj_rest_auth_version.split(".")[0]) < 3 and not cookie_name:
+            cookie_name = getattr(settings, 'JWT_AUTH_COOKIE', None)
+
```
[the remark this answers is not in the document]
Comment (nixsiow) [reply drf_spectacular/contrib/rest_auth.py:175]: Ah… Much more elegant than my solution. Sorry, I should have noticed it earlier, it's literally on the same file! Thanks for fixing this! :D

[the remark this answers is not in the document]
Comment (tfranzel) [reply drf_spectacular/contrib/rest_auth.py]: No problem, I totally forgot about it too. thanks again for the PR.

**Pull request merged**